src/FSharpLint.Client/FSharpLintToolLocator.fs

```diff
@@ -0,0 +1,256 @@
+                else 
+                    None)
+        | Error(ProcessStartError.ExecutableFileNotFound _)
+        | Error(ProcessStartError.UnexpectedException _) -> None)
+
+let findFSharpLintTool (workingDir: Folder) : Result<FSharpLintToolFound, FSharpLintToolError> =
+    // First try and find a local tool for the folder.
+    // Next see if there is a global tool.
+    // Lastly check if an executable is present on the PATH.
+    let localToolsListResult = runToolListCmd workingDir false
+
+    match localToolsListResult with
+    | Ok(CompatibleTool version) -> Ok(FSharpLintToolFound(version, FSharpLintToolStartInfo.LocalTool workingDir))
+    | Error err -> Error(FSharpLintToolError.DotNetListError err)
+    | Ok _localToolListResult ->
+        let globalToolsListResult = runToolListCmd workingDir true
+
+        match globalToolsListResult with
+        | Ok(CompatibleTool version) -> Ok(FSharpLintToolFound(version, FSharpLintToolStartInfo.GlobalTool))
+        | Error err -> Error(FSharpLintToolError.DotNetListError err)
+        | Ok _nonCompatibleGlobalVersion ->
+            let onPathVersion = fsharpLintVersionOnPath ()
+
+            match onPathVersion with
+            | Some(executableFile, FSharpLintVersion(CompatibleVersion version)) ->
+                Ok(FSharpLintToolFound((FSharpLintVersion(version)), FSharpLintToolStartInfo.ToolOnPath executableFile))
+            | _ -> Error FSharpLintToolError.NoCompatibleVersionFound
+
+let createFor (startInfo: FSharpLintToolStartInfo) : Result<RunningFSharpLintTool, ProcessStartError> =
+    let processStart =
+        match startInfo with
+        | FSharpLintToolStartInfo.LocalTool(workingDirectory: Folder) ->
+            ProcessStartInfo(
+                FileName = "dotnet",
+                WorkingDirectory = Folder.Unwrap workingDirectory,
+                Arguments = $"{FSharpLintToolName} --daemon")
+        | FSharpLintToolStartInfo.GlobalTool ->
+            let userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
+
+            let fsharpLintExecutable =
+                let fileName = if isWindows then $"{FSharpLintToolName}.exe" else FSharpLintToolName
+                Path.Combine(userProfile, ".dotnet", "tools", fileName)
+
+            ProcessStartInfo(
+                FileName = fsharpLintExecutable,
+                Arguments = "--daemon")
+        | FSharpLintToolStartInfo.ToolOnPath(FSharpLintExecutableFile executableFile) ->
+            ProcessStartInfo(
+                FileName = File.Unwrap executableFile,
+                Arguments = "--daemon")
+
+    processStart.UseShellExecute <- false
+    processStart.RedirectStandardInput <- true
+    processStart.RedirectStandardOutput <- true
+    processStart.RedirectStandardError <- true
+    processStart.CreateNoWindow <- true
+
+    match startProcess processStart with
+    | Ok daemonProcess ->
+        // fsharplint:disable-next-line RedundantNewKeyword
```

Comment:
@MrLuje why did you disable this rule several times in this PR? did you find a bug?

Reply:
Maybe ? It is `disposable`, so explicite `new` is recommanded (as a warning)
Without it, I have :
> Usage of `new` keyword here is redundant.F# Linter [FL0014](http://fsprojects.github.io/FSharpLint/how-tos/rules/FL0014.html)

Without the `new` :
> It is recommended that objects supporting the IDisposable interface are created using the syntax 'new Type(args)', rather than 'Type(args)' or 'Type' as a function value representing the constructor, to indicate that resources may be owned by the generated value F# Compiler [760](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-messages/fs0760)

Reply:
@MrLuje hey MrLuje, Mershad is right, see his branch (he edited his commit to add the link) where he removed the 2 comments disabling the rule and CI is still green

Reply:
Hum, probably my local setup then...
Just to make sure to understand properly, if I go to https://try.fsharp.org/ and type
```fsharp
module Tour.Functions

open System.IO
let s = StreamReader("")
``` 
I get the same warning :
```
It is recommended that objects supporting the IDisposable interface are created using the syntax 'new Type(args)', rather than 'Type(args)' or 'Type' as a function value representing the constructor, to indicate that resources may be owned by the generated value
```
Since `warnings as errors` is enabled in FSharpLint project, I expect the issue I locally have
https://github.com/fsprojects/FSharpLint/blob/08ceae74c1003f4b04ab274f92db758383ce399c/Directory.Build.props#L3-L4
Am i missing something ?

Reply:
@MrLuje not sure I understand what you're saying. You're bringing up a code snippet that uses a type which is IDisposable, therefore the `new` keyword is needed. That's why you get the warning (from F# compiler).

The rule RedundantNewKeyword is, though, for cases when you're using the `new` keyword in types that are **not** IDisposable. In those cases, using the keyword `new` is not needed. But if you do use it, you don't get a warning from the F# compiler telling you that you don't need it. That's why RedundantNewKeyword exists: to tell you this.

Did I miss something?

Reply:
@MrLuje anyway the strange thing here is that you're saying that if you use the `new` keyword, then you get an FSharpLint suggestion, and this is what Mersho refuted above: by showing you a commit whose CI is not red (which means our SelfCheck target passes with no errors; that is, FSharpLint rules are not flagging anything). So please just cherry-pick Mersho's commit from his branch here and push it to your branch.

Reply:
> @MrLuje not sure I understand what you're saying. You're bringing up a code snippet that uses a type which is IDisposable, therefore the `new` keyword is needed. That's why you get the warning (from F# compiler).
> 
> The rule RedundantNewKeyword is, though, for cases when you're using the `new` keyword in types that are **not** IDisposable. In those cases, using the keyword `new` is not needed. But if you do use it, you don't get a warning from the F# compiler telling you that you don't need it. That's why RedundantNewKeyword exists: to tell you this.
> 
> Did I miss something?

Nope, totally agree with this analysis 👍 
I noticed something else (and I may be missing something about how FSharpLint works) :
* running analysis on the project or solution doesn't trigger it
```bash
$ dotnet src/FSharpLint.Console/bin/Release/net6.0/dotnet-fsharplint.dll lint --file-type project src/FSharpLint.Client/FSharpLint.Client.fsproj
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/obj/Debug/net6.0/.NETCoreApp,Version=v6.0.AssemblyAttributes.fs ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/obj/Debug/net6.0/FSharpLint.Client.AssemblyInfo.fs ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/Contracts.fsi ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/Contracts.fs ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/LSPFSharpLintServiceTypes.fsi ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/LSPFSharpLintServiceTypes.fs ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/FSharpLintToolLocator.fsi ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/FSharpLintToolLocator.fs ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/LSPFSharpLintService.fsi ==========
========== Finished: 0 warnings ==========
========== Linting /home/vince/src/github/mrluje/FSharpLint.worktrees/rw/make_it_all_6/src/FSharpLint.Client/LSPFSharpLintService.fs ==========
========== Finished: 0 warnings ==========
========== Summary: 0 warnings ==========
```

* running it only on FSharpLintToolLocator.fs triggers it
```bash
$ dotnet src/FSharpLint.Console/bin/Release/net6.0/dotnet-fsharplint.dll lint src/FSharpLint.Client/FSharpLintToolLocator.fs 
========== Linting src/FSharpLint.Client/FSharpLintToolLocator.fs ==========
Usage of `new` keyword here is redundant.
Error on line 226 starting at column 22
        let handler = new HeaderDelimitedMessageHandler(
                      ^                                 
See https://fsprojects.github.io/FSharpLint/how-tos/rules/FL0014.html
--------------------------------------------------------------------------------
`x = true` might be able to be refactored into `x`.
Error on line 17 starting at column 53
        if supportedRange.IsSatisfied(parsedVersion, includePrerelease = true) then
                                                     ^                             
See https://fsprojects.github.io/FSharpLint/how-tos/rules/FL0065.html
--------------------------------------------------------------------------------
========== Finished: 2 warnings ==========
========== Summary: 2 warnings ==========
```
In any case, I agree with you there shouldn't be these rules disabled in the final code, I'm pushing Mersho's commit